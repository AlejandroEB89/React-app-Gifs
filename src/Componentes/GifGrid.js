// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {

    const { data:img , loading} = useFetchGifs( category );

    
    return ( 
        <>
            <h3 className="animate__animated animate__fadeIn"> { category } </h3>
            { loading && <p data-testid="pLoad" className="animate__animated animate__flash"> Cargando...</p> }
            <div className="card-grid">
            

                { img.map( imag => ( 
                    <GifGridItem 
                    key={ imag.id }
                    {...imag} />   //el op spread esta enviando el resto de propiedades que tiene el img como props al GifgridItem
                ) ) 
                }
            </div>
        </>
        
     );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
 
export default GifGrid;