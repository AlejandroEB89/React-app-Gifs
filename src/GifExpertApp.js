import { useState } from "react";
import AddCategory from "./Componentes/AddCategory";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Constelations', 'Universe', 'Planets'])
   

    //const handleAdd = () => {
      //  setCategories([...categories, "Atoms"]);
   // }

    return ( 
        <>
            <h2> Gif Expert App </h2>
            <AddCategory setCategories= { setCategories } />
            <hr />
           
            <ol>
                {categories.map( category => {
                    return <li key={category}> { category }</li>
                })}
            </ol> 
        </>
       
     );
}
 
export default GifExpertApp;