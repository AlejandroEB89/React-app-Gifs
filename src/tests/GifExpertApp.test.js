
import { render, screen } from "@testing-library/react";
import GifGrid from "../Componentes/GifGrid";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas en  <GifExpertApp />", () => {
    test('Debe renderizarse correctamente', () => {
        const { asFragment } = render( <GifExpertApp /> );
        expect( asFragment() ).toMatchSnapshot();
    })

    
})