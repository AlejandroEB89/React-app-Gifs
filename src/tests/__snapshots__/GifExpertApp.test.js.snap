// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifExpertApp /> Debe renderizarse correctamente 1`] = `
<DocumentFragment>
  <h2>
     GifSearchApp 
  </h2>
  <form
    name="form"
  >
    <input
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol>
    <h3
      class="animate__animated animate__fadeIn"
    >
       Universe 
    </h3>
    <p
      class="animate__animated animate__flash"
      data-testid="pLoad"
    >
       Cargando...
    </p>
    <div
      class="card-grid"
    />
  </ol>
</DocumentFragment>
`;
